src/sys/wait.rs

```diff
@@ -215,24 +236,94 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
-    use self::WaitStatus::*;
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+/// Designates whether the supplied `Pid` value is a process ID, process group ID,
+/// specifies any child of the current process's group ID, or any child of the current process.
+pub enum PidGroup {
+    ProcessID(u32),
+    ProcessGroupID(u32),
+    AnyGroupChild,
+    AnyChild,
+}
+
+impl From<Pid> for PidGroup {
```

Comment:
The `From`s for `PidGroup` should be removed in favor on embedded `Pid`s directly into the specific enum types.